### PR TITLE
CI: Updated version of macos image

### LINKED ISSUE #31281: CI: update azure VM image
Email from Azure support

> On March 23, 2020, we’ll be removing the following Azure Pipelines hosted images:
>
> * Windows Server 2012R2 with Visual Studio 2015, a.k.a vs2015-win2012r2
> * macOS X High Sierra 10.13, a.k.a macOS-10.13
> * Windows Server Core 1803, a.k.a win1803

> If you use YAML pipelines, update your pipeline by editing its YAML file.
> 
>  
> 
>   jobs:
> 
>     - job: Windows
> 
>       pool:
> 
>         vmImage: 'vs2017-win2016'
> 
>       steps:
> 
>       - script: echo hello from Windows

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] closes #31281
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ jobs:
 - template: ci/azure/posix.yml
   parameters:
     name: macOS
-    vmImage: xcode9-macos10.13
+    vmImage: macos10.15
```
Comment (WillAyd) [inline azure-pipelines.yml]: FWIW I would be OK with `macos-latest` - I don't see a need to pin a particular version

Comment (WillAyd) [reply azure-pipelines.yml]: But I think failure is being 10.15 isn't available via pipelines:

https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent

Comment (MomIsBestFriend) [reply azure-pipelines.yml]: > FWIW I would be OK with `macos-latest` - I don't see a need to pin a particular version

Though so as well, but I figured that ```xcode9-macos10.13``` is there for a reason, apparently I was wrong

**Comment by MomIsBestFriend** [comment]
Will #30990 work after this change?

does ```gcc``` support macos 10.15?

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by WillAyd** [comment]
Hmm though looks like the macOS build is stuck in queued

**Comment by MomIsBestFriend** [comment]
> Hmm though looks like the macOS build is stuck in queued

@WillAyd Not anymore, I guess it was downloading the image or refreshing the cache.

File: azure-pipelines.yml (hunk 1 above)
Comment (gfyoung) [reply azure-pipelines.yml]: Using latest is a little dicey IMO.

You don't know what incompatibilities newer versions may introduce.

Comment (gfyoung) [reply azure-pipelines.yml]: Using latest is a little dicey IMO.

You don't know what incompatibilities newer versions may (unexpectedly) introduce. I think pinning to the latest (numeric) version as was done before is best.

I don't necessarily object to testing against macos-latest. However, I would mark that build as an allowable failure (though not sure how one does that with Azure pipelines).

**Review by gfyoung — CHANGES_REQUESTED** [review]
Using `latest` is a little dicey IMO.

You don't know what incompatibilities newer versions may (unexpectedly) introduce. I think pinning to the latest (numeric) version as was [done](https://github.com/pandas-dev/pandas/pull/31292#discussion_r370787740) before is best.

I don't necessarily object to testing against `macos-latest`. However, I would mark that build as an allowable failure (though not sure how one does that with Azure pipelines).

**Comment by jreback** [comment]
seems reasonable

**Pull request merged**